src/Microsoft.OpenApi.OData.Reader/Generator/OpenApiParameterGenerator.cs

```diff
@@ -216,6 +216,56 @@ public static IList<OpenApiParameter> CreateParameters(this ODataContext context
             return parameters;
         }
 
+        /// <summary>
+        /// Creates the path parameters for the <see cref="ODataPath"/>
+        /// </summary>
+        /// <param name="path">The ODataPath</param>
+        /// <param name="context">The OData context.</param>
+        /// <returns>The created list of <see cref="OpenApiParameter"/></returns>
+        public static List<OpenApiParameter> CreatePathParameters(this ODataPath path, ODataContext context)
+        {
+            List<OpenApiParameter> pathParameters = new();
+            var parameterMappings = path.CalculateParameterMapping(context.Settings);
+
+            foreach (ODataKeySegment keySegment in path.OfType<ODataKeySegment>())
+            {
+                IDictionary<string, string> mapping = parameterMappings[keySegment];
+                pathParameters.AddRange(context.CreateKeyParameters(keySegment, mapping));
+            }
+
+            // Add the route prefix parameter v1{data}
+            if (context.Settings.RoutePathPrefixProvider?.Parameters != null)
+            {
+                pathParameters.AddRange(context.Settings.RoutePathPrefixProvider.Parameters);
+            }
+
+            return pathParameters;
```

Comment:
shouldn't this also add the odata functions parameters? e.g. `/GetSalesTaxRate(PostalCode=10)`

Reply:
There is no benefit of using Path Parameters for OData Actions and Functions.  Even composable functions.